spec/03-plugins/11-basic-auth/02-api_spec.lua

```diff
@@ -301,4 +302,104 @@ describe("Plugin: basic-auth (API)", function()
+        })
+      end)
+      it("retrieve all the basic-auths", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/basic-auths/"
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(2, #json.data)
+        assert.equal(2, json.total)
+      end)
+      it("retrieve all the basic-auths", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/basic-auths"
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(2, #json.data)
+        assert.equal(2, json.total)
+      end)
+      it("retrieve basic-auths for a consumer_id", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/basic-auths?consumer_id=" .. consumer.id
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(1, #json.data)
+        assert.equal(1, json.total)
+      end)
+      it("return empty for a non-existing consumer_id", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/basic-auths?consumer_id=" .. utils.uuid(),
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(0, #json.data)
+        assert.equal(0, json.total)
+      end)
+    end)
+  end)
+  describe("/basic-auths/:credential_username_or_id/consumer", function()
+    describe("GET", function()
+      local credential
+      setup(function()
+        helpers.dao:truncate_table("basicauth_credentials")
+        credential = assert(helpers.dao.basicauth_credentials:insert {
+          consumer_id = consumer.id,
+          username = "bob"
+        })
+      end)
+      it("retrieve consumer from a basic-auth id", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/basic-auths/" .. credential.id .. "/consumer"
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.same(consumer,json)
+      end)
+      it("retrieve consumer from a basic-auth username", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/basic-auths/" .. credential.username .. "/consumer"
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.same(consumer,json)
+      end)
+      it("returns 404 for a random non-existing basic-auth id", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/basic-auths/" .. utils.uuid()  .. "/consumer"
```

Comment:
Since we modified the parameter to be `:credential_username_or_id`, we should also add a test case for a non-existing username this test case only covers non-existing id).

Reply:
Done